docs/source/en/hf_quantizer.md

```diff
@@ -0,0 +1,67 @@
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Build a new `HFQuantizer` class to add quantization support for a new quantization method.
+
+Through this document, you will learn how to work on a transformers integration of a new quantization method. Note that currently the `HFQuantizer` is not meant to be used for any PyTorch module, but you should rather see it as an internal utility class that is used in the core modeling code to easily quantize transformers models with different SoTA approaches (e.g. QLoRA, GPTQ, LLM.int8, AWQ, ...). 
+
+
+## Pre-requisities 
+
+Before you start integrating a new quantization method into transformers, make sure that the method you are trying to add meet the following pre-requisities. Note we only support quantization methods that can be run with PyTorch modules for now.
+
+- The quantization method is available through a Python package that is pip-installable by anyone (it is also fine if you can only install the package from source), ideally with pre-compiled kernels included in the pip package.
+- The method can at least run on a commonly-used hardware (CPU, GPU, ..).
+- The method is wrapped in a `nn.Module` (e.g. `Linear8bitLt`, `Linear4bit`). Ideally your quantized linear layer should have the following definition
+```py
+class Linear4bit(nn.Module):
+    def __init__(self, ...):
+        ...
+    
+    def forward(self, x):
+        return my_4bit_kernel(x, self.weight, self.bias)
+```
+That way, transformers models can be easily quantizable by simply replacing some instances of `nn.Linear` with a target class.
+- Ideally the quantization method should be serializable, i.e. you can save the quantized weights locally or push them on the Hub.
+- Make sure the package that contains the quantization kernels / primitive is mature enough (e.g. no frequent breaking changes).
+
+Note that for some quantization methods it is a strong requirement to "pre-quantize" the models through data calibration (e.g. AWQ). In that case, we prefer to support only inference through transformers and let third-party libraries maintained by the ML community deal with the model quantization itself.
+
+## How should I get started?
+
+0- 📕 Create a new quantization config class inside `src/transformers/utils/quantization_config.py`, and make sure to expose that new quantization config inside transformers main init, by adding it on the `_import_structure` object of `src/transformers/__init__.py`.
```

Comment:
I think its more common to start a list with "1" (will need to shift the following list items). It may also be a better user experience to add a link to the specific object (`_import_structure`) referenced so users don't spend as much time looking for it.

```suggestion
1. 📕 Create a new quantization config class inside `src/transformers/utils/quantization_config.py` and make sure to expose the new quantization config inside Transformers main `init` by adding it to the `_import_structure` object of `src/transformers/__init__.py`.
```